src/ORM/DataObject.php

```diff
@@ -1238,6 +1238,15 @@ public function forceChange()
     public function validate()
     {
         $result = ValidationResult::create();
+        // Call DBField::validate() on every DBField
+        $specs = static::getSchema()->fieldSpecs(static::class);
+        foreach (array_keys($specs) as $fieldName) {
+            $dbField = $this->dbObject($fieldName);
+            $validationResult = $dbField->validate();
+            if (!$validationResult->isValid()) {
+                $result->combineAnd($validationResult);
+            }
```

Comment:
```suggestion
            $result->combineAnd($dbField->validate());
```
If you do the check, and the validation result has some non-error message, that non-error message gets discarded.

Reply:
I don't understand what this means

What's a non-error message in the context of validation?

Reply:
Any message that when set, doesn't set the validation result to invalid (e.g. warnings)